src/wallet/mod.rs

```diff
@@ -274,6 +308,12 @@ where
 				},
 			}
 
+			let mut locked_persister = self.persister.lock().unwrap();
+			locked_wallet.persist(&mut locked_persister).map_err(|e| {
+				log_error!(self.logger, "Failed to persist wallet: {}", e);
+				Error::PersistenceFailed
+			})?;
```

Comment:
If `psbt.extract_tx()` fails below, is there any implications of persisting here? Will the wallet think the inputs have been spent? Will it think a change output has been used when it really hasn't?

Reply:
If we never call `take_staged`, does that mean the `ChangeSet` returned by `staged` will continuously grow. I assume the purpose of it is to communicate what needs to be persisted, but it wasn't 100% clear from the docs. It mentions "committed" in some paces and "persisted" in others, though I assume those are equivalent?

Reply:
Ah, so `take_staged` is basically "unstage".